src/Microsoft.OData.Client/BaseSaveResult.cs

```diff
@@ -247,7 +247,7 @@ internal static DataServiceClientException GetResponseText(Func<Stream> getRespo
 
             using (Stream stream = getResponseStream())
             {
-                if ((null != stream) && stream.CanRead)
+                if ((null != stream) && stream.CanRead && stream.CanSeek)

```

Comment:
I think another alternative would be to simply pass along the clone memory stream to the odata message reader, so that you don't have to bother checking whether the original stream is seekable

e.g.
```c#
message = new StreamReader(cloneStream).ReadToEnd();
cloneStream.Position = 0;
Func<Stream> getCloneResponseStream = () => cloneStream;
// ...
httpWebResponseMessage  = new HttpWebResponseMessage(headers, (int) statusCode, getCloneResponseStream);
```

But if the original `getResponseStream` delegate always returns a new stream instance, then maybe there's no need in going through rewinding or cloning in the first place. But I don't know if that's the case.

Reply:
getResponseStream  Returns the same instance. The current design of Odata message classes does not allow passing the stream object just like that.

Reply:
But the `HttpWebResponseMessage` does not care whether it's the same instance. So if you passed a delegate that returns a different/cloned stream with the same data, it should still work I believe.

Regarding the design of odata messages, it seems like a bit of hack to create a `HttpWebResponseMessage` just so that you can parse the error out of the stream. Maybe this could be something worth refactoring down the line.